identity_credential/src/presentation/presentation.rs

```diff
@@ -38,8 +38,9 @@ pub struct Presentation<CRED, T = Object> {
   #[serde(rename = "type")]
   pub types: OneOrMany<String>,
   /// Credential(s) expressing the claims of the `Presentation`.
-  #[serde(default = "Default::default", rename = "verifiableCredential")]
-  pub verifiable_credential: OneOrMany<CRED>,
+  #[rustfmt::skip]
+  #[serde(default = "Default::default", rename = "verifiableCredential", skip_serializing_if = "Vec::is_empty")]
+  pub verifiable_credential: Vec<CRED>,
```

Comment:
Thanks for opening the PR!

With this change we would have at least these differences to before:
1. When deserializing, an empty array in JSON would be allowed.
2. When constructing a presentation, an empty vec would be allowed to be set, which it was previously not.

Case 2 is fine, because that's our "new way" to serialize a presentation without the `verifiableCredential` field, which we probably have no use case for, but can allow without harm.

Case 1 I think we should disallow with a `deserialize_with = "path"`. We implement the VC Data Model 1.1 and it says:
> If present, the value of the verifiableCredential [property](https://www.w3.org/TR/vc-data-model/#dfn-property) MUST be constructed from one or more [verifiable credentials](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-credentials)

I interpret that as: If the field is present, then it must be a non-empty array.

@nanderstabel Do you have any thoughts on that?

Reply:
Yes I agree with you! Disallowing an empty array value is more in line with the spec so I took you suggestion and added
`deserialize_verifiable_credential`. 

I also added a new corresponding `Error` variant: `EmptyVerifiableCredentialsArray`. Not sure whether it adheres to any
naming convention you may have for the different Errors so please let me know if you would like me to change it!